### PR TITLE
Create get repository API 

### PR DESCRIPTION
**Pull request by AsafCX**
### Description

As part of generating webhook key flow, A new API get repository added.

### References

https://checkmarx.atlassian.net/browse/AST-284

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).  *If documentation is a Wiki Update, please indicate desired changes within PR MD Comment*
- [X] All active GitHub checks for tests, formatting, and security are passing
- [X] The correct base branch is being used

### CODE REVIEW

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 60 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**0 Medium**<br />Low&nbsp;**56 Low**<br />Info&nbsp;**4 Info**<br />
### Violation Summary
#### No policy violation found
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
***
Logo
**Checkmarx SCA - Scan Summary & Details**
### Cx-SCA Summary
**Total Packages Identified**: **102**
**Scan Risk Score**: **9.80**

High&nbsp;**2** **High** **severity vulnerabilities**
Medium&nbsp;**0** **Medium** **severity vulnerabilities**
Low&nbsp;**0** **Low** **severity vulnerabilities**
[View more details on Checkmarx UI](https://eu.sca.checkmarx.net/#/projects/5d7c94ee-6ff1-4535-be49-451a382b1e6b/reports/c41e60aa-ffe3-4058-95eb-c82fdf28e39f)
### Cx-SCA vulnerability result overview
|Vulnerability ID|Package|Severity|CVSS score|Publish date|Current version|Recommended version|Link in CxSCA|Reference – NVD link|
---|---|---|---|---|---|---|---|---
`CVE-2016-1000027`|org.springframework:spring-web|HIGH|9.8|2020-01-02T23:15:00|5.3.3|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/5d7c94ee-6ff1-4535-be49-451a382b1e6b/reports/c41e60aa-ffe3-4058-95eb-c82fdf28e39f/vulnerabilities/CVE-2016-1000027%3AYarn-org.springframework%3Aspring-web-5.3.3/vulnerabilityDetails)|[CVE-2016-1000027](https://nvd.nist.gov/vuln/detail/CVE-2016-1000027)
`CVE-2016-1000027`|org.springframework:spring-webmvc|HIGH|9.8|2020-01-02T23:15:00|5.3.3|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/5d7c94ee-6ff1-4535-be49-451a382b1e6b/reports/c41e60aa-ffe3-4058-95eb-c82fdf28e39f/vulnerabilities/CVE-2016-1000027%3AYarn-org.springframework%3Aspring-webmvc-5.3.3/vulnerabilityDetails)|[CVE-2016-1000027](https://nvd.nist.gov/vuln/detail/CVE-2016-1000027)

File: src/main/java/com/checkmarx/controller/WebController.java [hunk 1]
```diff
@@ -71,20 +70,20 @@
...
-       RepoDto repoDto = scmService.getScmOrgRepo(orgId, repoId);
-        log.info("getRepository result: {}", repoDto);
+    @GetMapping(value = "orgs/{orgIdentity}/repos/{repoIdentity}/search")
+    public ResponseEntity<RepoDto> getRepositoryByRepoUrl(@PathVariable @Pattern(regexp = VALIDATION_REGEX) String orgIdentity,
+                                                          @PathVariable @Pattern(regexp = VALIDATION_REGEX) String repoIdentity,
+                                                          @RequestParam String repoUrl) {
+        log.trace("getRepositoryByRepoUrl: orgIdentity={}, repoIdentity={}, repoUrl={}", orgIdentity, repoIdentity, repoUrl);
+        ScmService scmService = scmServiceFactory.getInstanceByRepoUrl(repoUrl);
+        RepoDto repoDto = scmService.getScmOrgRepo(orgIdentity, repoIdentity);
+        log.info("getRepositoryByRepoUrl result: {}", repoDto);
```
Comment (alex-ko-dev) [inline src/main/java/com/checkmarx/controller/WebController.java:86]: Here webhook key will be written into log as plain text, which is probably not a good idea.

Comment (AsafCX) [reply src/main/java/com/checkmarx/controller/WebController.java:86]: Done.

**Review by NimrodGolan — APPROVED** [review]

**Pull request merged**